### PR TITLE
[ROCm] Upstream rocm fixes

### PR DESCRIPTION
**Pull request by rsanthanam-amd**
/cc @cheshire

### CODE REVIEW

**Review by cheshire — APPROVED** [review]

**Comment by rsanthanam-amd** [comment]
This PR is failing to merge because of CI failures but I am unable to ascertain what the problem is.  Can someone take a look?

**Comment by cheshire** [comment]
Why are you changing code in tensorflow/lite?

**Comment by rsanthanam-amd** [comment]
Our Ubuntu-sanity test failed if I didn't remove those dependencies.   The conv_buffer_1x1 stuff was removed in some commit but it appears that commit failed to remove those [stale] refences in tf-lite.

**Review by akuegel — COMMENTED** [review]
The dependency changes from this PR seem to be causing failures, and I think nobody understands why.

File: tensorflow/compiler/xla/service/gpu/amdgpu_compiler.cc [hunk 1]
```diff
@@ -95,17 +95,7 @@ Status AMDGPUCompiler::OptimizeHloConvolutionCanonicalization(
...
-  // When transposes appear in a fusion node, we can easily adjust the
-  // multi-dimensional index to create the one needed for the operand. This
-  // is not as easy with bitcasts, because we don't have the information
-  // readily available which dimensions are permuted. In addition to that,
-  // if we have a transpose and a reshape next to each other, they will both
-  // be replaced by a bitcast, and we replace bitcast(bitcast) with one
-  // bitcast. This leads to having to linearize and then delinearize the
-  // index.
   AlgebraicSimplifierOptions options;
-  options.set_replace_transpose_with_bitcast(false);
```
[the remark this answers is not in the document]
Comment (rsanthanam-amd) [reply tensorflow/compiler/xla/service/gpu/amdgpu_compiler.cc:108]: I'll pick up your changes and just close this PR.

**Comment by rsanthanam-amd** [comment]
Closing because this is obsolete and causing CI issues.

**Pull request closed without merging**